geopatterns_demo/file_handler.py

```diff
@@ -0,0 +1,74 @@
+from io import BytesIO
+
+import boto3
+
+from geopatterns_demo import models, settings
+from geopatterns_demo.draw import geopattern
+
+
+def hash_name(text: str) -> str:
+    """Generate the file name using the hash function.
+
+    Args:
+        text (str): text that we get from get request
+
+    Returns:
+        str: file name
+    """
+    return 'image{0}.png'.format(hash(text))
+
+
+def check_file(file_name: str, bucket: boto3) -> bool:
+    """Check the file for presence in the S3 bucket.
+
+    Args:
+        file_name (str): image file name
+        bucket (boto3): boto3 object - S3 bucket
+
+    Returns:
+        bool: bool
+    """
+    bucket_object = list(bucket.objects.filter(Prefix=file_name))
+    if len(bucket_object) > 0:  # noqa: WPS507
+        return bucket_object
+
+
+def file_to_s3(file_name: str, image: BytesIO) -> str:
+    """Save file to S3 bucket & return it's URL.
+
+    Args:
+        file_name (str): image file name
+        image (BytesIO): image object in bytes
+
+    Returns:
+        str: url
+    """
+    s3 = boto3.Resource('s3')
+    bucket = s3.Bucket(settings.S3_BUCKET_NAME)
+    bucket.upload_fileobj(image.read(), file_name)
+    if check_file(file_name, bucket):
```

Comment:
This probably will not work.

S3 is a storage that is eventually consistent. Eventual consistency means that, once you've uploaded a file to S3, it will appear there and will be available for retrieval but that is not guaranteed to happen momentarily.

There might be a delay — I do not know how long but, say, 0.5 sec — before the file appears. So, this `check_file()` call can easily miss the file because it is executed without a delay.

Do we actually **need** to check the existence of the uploaded file here?

I think not. S3 is a well tested system. We just need to upload the file and send the URL to the user, that is it.